app/bundles/EmailBundle/Mailer/Transport/AbstractTokenArrayTransport.php

```diff
@@ -0,0 +1,192 @@
+use Mautic\EmailBundle\Mailer\Message\MauticMessage;
+use Symfony\Component\Mailer\Exception\TransportException;
+use Symfony\Component\Mailer\Header\MetadataHeader;
+use Symfony\Component\Mailer\Header\TagHeader;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\AbstractTransport;
+
+/**
+ * The goal of this class is to provide a common interface for transports that support tokenized sending.
+ * there are abstract functions that are inherited from the AbstractTransport class that must be implemented.
+ */
+abstract class AbstractTokenArrayTransport extends AbstractTransport implements TokenTransportInterface
+{
+    /**
+     * @var MauticMessage
+     */
+    protected $message;
+
+    /**
+     * @var array<string>
+     */
+    protected $standardHeaderKeys = [
+        'MIME-Version',
+        'received',
+        'dkim-signature',
+        'Content-Type',
+        'Content-Transfer-Encoding',
+        'To',
+        'From',
+        'Subject',
+        'Reply-To',
+        'CC',
+        'BCC',
+    ];
+
+    abstract protected function doSend(SentMessage $message): void;
+
+    /**
+     * Get the metadata from a MauticMessage.
+     *
+     * @return array<string, mixed>
+     */
+    public function getMetadata()
+    {
+        return $this->message->getMetadata();
+    }
+
+    /**
+     * Replace the tokens in the message with their real values.
+     *
+     * @param array<string> $search  If the mailer requires tokens in another format than Mautic's, pass array of Mautic tokens to replace
+     * @param array<string> $replace If the mailer requires tokens in another format than Mautic's, pass array of replacement tokens
+     */
+    protected function replaceTokens($search = [], $replace = []): void
+    {
+        if (!empty($search)) {
+            MailHelper::searchReplaceTokens($search, $replace, $this->message);
+        }
+    }
+
+    /**
+     * Converts Mime\Email into associative array.
+     * This function is kept for backward compatibility.
+     *
+     * @param array<string> $search            If the mailer requires tokens in another format than Mautic's, pass array of Mautic tokens to replace
+     * @param array<string> $replace           If the mailer requires tokens in another format than Mautic's, pass array of replacement tokens
+     * @param bool|false    $binaryAttachments True to convert file attachments to binary
+     *
+     * @return array<string, mixed>
+     */
+    protected function messageToArray($search = [], $replace = [], $binaryAttachments = false)
+    {
+        if (!empty($search)) {
+            MailHelper::searchReplaceTokens($search, $replace, $this->message);
+        }
+
+        $from = $this->message->getFrom();
+        foreach ($from as $address) {
+            $fromEmail = $address->getEncodedAddress();
+            $fromName  = $address->getEncodedName();
+        }
+
+        $message = [
+            'html'    => $this->message->getTextBody(),
```

Comment:
```suggestion
            'html'    => $this->message->getHtmlBody(),
```
@mabumusa1 is `'html' => $this->message->getTextBody()` done by design or is it supposed to be the way I code suggest.

Reply:
@fedys your suggestion is correct, I think it is a mistake to have the `textBody` there